spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/DistriOptimizer.scala

```diff
@@ -677,6 +677,12 @@ class DistriOptimizer[T: ClassTag] (
 
   private var models: RDD[DistriOptimizer.Cache[T]] = null
 
+  import DistriOptimizer._
+  if (dataset.originRDD != null) {
+    logger.info("Loading and caching data ...")
+    dataset.originRDD().cache().count()
```

Comment:
My origin thought is to avoid user created his own rdd and didn't cache it. Since the optimizer use dataset.originRDD to training, what about we cache dataset.originRDD here and remove cache operation in dataset.scala.